test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -374,6 +467,27 @@ void validateResults(ReaderGroupManager readerGroupManager, String readerGroupNa
         readerGroupManager.deleteReaderGroup(readerGroupName);
     }
 
+    void waitForScaling(String scope) throws InterruptedException, ExecutionException {
+        for (int waitCounter = 0; waitCounter < 12; waitCounter++) {
+            StreamSegments streamSegments = controller.getCurrentSegments(scope, AUTO_SCALE_STREAM).get();
+            testState.currentNumOfSegments.set(streamSegments.getSegments().size());
+            if (testState.currentNumOfSegments.get() == 2) {
+                log.info("The current number of segments is equal to 2, ScaleOperation did not happen");
+                //Scaling operation did not happen, wait
+                Exceptions.handleInterrupted(() -> Thread.sleep(10000));
+                throw new AbstractFailoverTests.ScaleOperationNotDoneException();
```

Comment:
Where are we using ScaleOperatonNotDoneException ?

Reply:
we are defining it here, https://github.com/pravega/pravega/pull/1921/files#diff-7c588e31332a0f89d89d04375f736605L423

Reply:
if the scale does not happen then we need to retry right?
In the current current code we will exit the `for `loop (with loop counter 12)

Reply:
as we are not using it else where, removed it.

Reply:
added a constant `SCALE_WAIT_ITERATIONS = 12;`